lib/tasks/assets.rake

```diff
@@ -0,0 +1,89 @@
+namespace :assets do
+
+  task :download_and_precompile, [:url_template, :credentials, :branch, :source_assets_path] => :environment do |t, args|
+    if args[:url_template].blank?
+      puts "Not including any external assets"
+      next
+    end
+
+    target_path = args[:source_assets_path] || "./tmp/assets_source"
+    FileUtils.mkdir_p target_path
+    Rake::Task['assets:download_external_assets'].invoke(target_path, args[:url_template], args[:credentials], args[:branch])
+    Rake::Task['assets:precompile_assets'].invoke(target_path)
+  end
+
+  # Downloads a tar from a URL, extracts its contents into <target_path>
+  #
+  # Params:
+  # * target_path: the dir path where the contents of the downloaded and extracted files will be put
+  # * url: any http[s] url. Optionally it can include a tag <branch> that will be replaced with
+  #   the contents of the branch param.
+  # * credentials: http basic auth credentials. The format should be user:password.
+  # * branch: Mainly for Github URLs usage. The content of this param will replace the <branch> tag
+  #   in the url param.
+  #
+  # Example:
+  #  rake deploy:precompile_assets["https://api.github.com/repos/organisation/repo/tarball/<branch>","deploy-user:secret","master"]
+  desc "Download external assets"
+  task :download_external_assets, [:target_path, :url_template, :credentials, :branch] => :environment do |t, args|
+    target_path = args[:target_path]
+    url_template = args[:url_template]
+    credentials = args[:credentials]
+    current_branch = args[:branch]
+    tar_file_path = "./tmp/assets.tar"
+
+
+    if url_template.blank? || target_path.blank?
+      raise 'usage: rake deploy:download_external_assets[target_path,url_template[,"user:password"][,branch]]'
+    end
+
+    FileUtils.mkdir_p target_path
+
+    urls = [current_branch, Settings.default_asset_branch, 'master'].map do |branch|
+      branch.present? ? url_template.gsub("<branch>", branch) : nil
+    end.compact
+
+    # Set github credentials --------------------------
+    http_options = {}
+    if credentials.present?
+      u, p = credentials.split(":")
+      http_options[:basic_auth] = { username: u, password: p }
+    end
+
+    # Download tar file --------------------------
+    errors, response = [], nil
+    urls.each do |url|
+      puts "Downloading external assets from #{url}"
+      response = HTTParty.get url, http_options
+      break if response.success?
```

Comment:
Nope - `urls` is essentially a re-try list in case of failure. It first tries to fetch for the current branch, then it tries fetches from whatever branch you set as the default in settings, then it tries to fetch from master. Once it has the assets from one branch, there's no point downloading from a different one.